### ISSUE TITLE
Alerts stopped working

### ISSUE THREAD

**Issue by robimarko** [librenms/librenms#6742]
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

GitHub issues is for  bugs, please do not post issues asking for help or how to do X, Y or Z. 
You can use our irc channel ##librenms on freenode to ask questions or our [community site](https://community.librenms.org).
If you have a feature request please post this on our [community site](https://community.librenms.org/c/feature-requests).

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [X] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [X] Please include all of the information between the `====================================` section of `./validate.php` which you can run from the cli.
- [X] Please provide ALL info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
- [X] Please provide as much detail as possible.
- [X] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

Hi,
I was using LibreNMS for monitoring community mesh without issues until a week ago.
Then suddenly we stopped receiving email and Rocket chat alerts.
I have searched around you PR-s and googled but no solution has been found.

I first ran validate.php and it found a couple of issues with DB, most likely due to the power outage and sudden shutdown.

I backed up the installation and repaired database with commands from validate.php and run it again to verify that no errors have been found.
Then I manually started daily.sh to ensure that latest code is used and rerun validate.php.
No errors have been returned.
validate.php returns this info:
==========================================================
Component | Version
--------- | -------
LibreNMS  | 6a79d4495f1a5d91d476021b7fb3ab58889e6f80
DB Schema | 192
PHP       | 7.0.18-0ubuntu0.16.04.1
MySQL     | 10.0.29-MariaDB-0ubuntu0.16.04.1
RRDTool   | 1.5.5
SNMP      | NET-SNMP 5.7.3
==========================================================

[OK]    Database connection successful
[OK]    Database schema correct

After that I manually run alerts.php -d and output that into log file.
It outputs this:
https://gist.github.com/robimarko/5e573d18ccc88e325076ba0ba5732dc2
I have tested mail -s manually and it is able to send emails, also rocket chat works when test button is used.
It looks like alert is not issued to transport according to eventlog.

I have exhausted my knowledge so far.

**Comment by laf** [comment]
GitHub a for bug reports and new device requests. Please open a thread on the community forum or use irc

**Comment by wdst-gcet** [comment]
We have noticed something similar to this. Our alerts have stoped sending out Via transports for Email and slack. However when someone acknowledges an alert all the transports send out an acknowledgment. We don't know when this started for us but It's been driving us nuts. We have remove all our custom alert templates and still have had no luck.

**Comment by robimarko** [comment]
Have you tried stopping alerts cronjob and then manually running it with -d
flag?
That way you can see the output for debugging

**Comment by wdst-gcet** [comment]
I gave that a shot and I do see all the alerts. It looks like the alerts show in the database. The alerts just wont seem to send the Email or slack Transport. When we run the the alert capture from the Web UI it is showing that it is sending a Transport: mail but it wont send  

[alerts.txt](https://github.com/librenms/librenms/files/1053591/alerts.txt)

**Comment by robimarko** [comment]
What does validate.php output?

**Comment by wdst-gcet** [comment]
We are running with a distributed poller so we are checking some other potential issues. What's strange is that this morning we received 100+ Emails and Slack alerts about 2 hours after the port went down, I have attached and image of the event log. Could this maybe be an issue with how a CRON job is running? The alert is seen with in minutes. But the transport doesn't send off for a few hours. 

I have validate.php 


[validate.txt](https://github.com/librenms/librenms/files/1055000/validate.txt)

 
![delayed-transport-send](https://cloud.githubusercontent.com/assets/20130657/26828958/9cd6a2ac-4a91-11e7-8035-43bb5f632a97.png)

**Comment by robimarko** [comment]
Hm,this looks like transport is the issue since alert is issued to
transport.
Can you manually run transport php file with -d flag?

**Comment by robimarko** [comment]
Also can you try to manually send email to verify that that is working

**Comment by wdst-gcet** [comment]
I just tested the Email and Slack Transports from the Web UI and the test Email and Slack alert both get sent. As for the testing of the transport php file are you referring to the files located in `/opt/librenms/includes/alerts`. I just want to make sure I am testing from the right folders.

**Comment by robimarko** [comment]
Should be that file.
It looks like this is issue with you template.
You can test your template.
There is info on that in docs

**Comment by wdst-gcet** [comment]
I ran a test on the template and rule with `./test-template.php -t 7 -d -h gnf-pop1-agn1-1 -r 36` and it looks like it's working correctly I think... I get the following results 

[template-test.txt](https://github.com/librenms/librenms/files/1055121/template-test.txt)

**Comment by robimarko** [comment]
Yeah,it looks normal.
This is really some weird reason that you are not getting notifications.

**Comment by wdst-gcet** [comment]
I don't know if it is us but we just did a new install with a new database etc and could only get (Alert acknowledge) Transports to send out via Email. The alerts show under /alerts/ but we never see a transport sent in the device event log. I think there is something else going on here. But if no one else is having this issue it must be something that we are doing wrong.

**Comment by robimarko** [comment]
Can you go to eventlog and then select NULL?

**Comment by wdst-gcet** [comment]
Okay so, we feel silly. After looking over all the configs this weekend we noticed that when we built a new database in AWS using RDS the time zone on the Data base and the server were off. The RDS instance was using UTC and the librenms server was using `"America/New_York"`. I think setting any other time zone then UTC on server can become a problem. After changing the server to UTC and the PHP setting to UTC all worked great! 

Is there a way for users to set there own time zone in the web UI? The only reason why I ask is because the added time conversion when parsing through the event log may not be the best idea for some network ops engineers. Especially durning a massive outage or in an emergency. Obviously the conversion is not hard to do. Some people may find it hard under a network wide outage.
